java/src/test/java/ai/rapids/cudf/DecimalColumnVectorTest.java

```diff
@@ -0,0 +1,187 @@
+    null,
+    BigDecimal.valueOf(rdSeed.nextLong(), 10),
+  };
+
+  @Test
+  public void testCreateColumnVectorBuilder() {
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL32, -5), 3,
+        (b) -> b.append(BigDecimal.valueOf(123456789, 5)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL64, -10), 3,
+        (b) -> b.append(BigDecimal.valueOf(1023040506070809L, 10)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+  }
+
+  @Test
+  public void testUpperIndexOutOfBoundsException() {
+    try (HostColumnVector decColumnVector = HostColumnVector.fromBigDecimals(decimal32Zoo)) {
+      assertThrows(AssertionError.class, () -> decColumnVector.getBigDecimal(3));
+      assertFalse(decColumnVector.hasNulls());
+    }
+  }
+
+  @Test
+  public void testLowerIndexOutOfBoundsException() {
+    try (HostColumnVector doubleColumnVector = HostColumnVector.fromBigDecimals(decimal32Zoo)) {
+      assertFalse(doubleColumnVector.hasNulls());
+      assertThrows(AssertionError.class, () -> doubleColumnVector.getBigDecimal(-1));
+    }
+  }
+
+  @Test
+  public void testAddingNullValues() {
+    try (HostColumnVector cv = HostColumnVector.fromBigDecimals(decimal64Zoo)) {
+      assertTrue(cv.hasNulls());
+      assertEquals(1, cv.getNullCount());
+      assertFalse(cv.isNull(0));
+      assertFalse(cv.isNull(1));
+      assertTrue(cv.isNull(2));
+      assertFalse(cv.isNull(3));
+    }
+  }
+
+  @Test
+  public void testOverrunningTheBuffer() {
+    try (Builder builder = HostColumnVector.builder(DType.create(DType.DTypeEnum.DECIMAL32, 4), 3)) {
+      assertThrows(AssertionError.class,
+          () -> builder.append(decimal32Zoo[0]).appendNull().appendBoxed(decimal32Zoo[1], decimal32Zoo[2]).build());
+    }
+  }
+
+  @Test
+  public void testDecimalValidation() {
+    assertThrows(AssertionError.class,
+        () -> HostColumnVector.fromBigDecimals(BigDecimal.valueOf(123, 1), BigDecimal.valueOf(123, 2)));
+    assertThrows(AssertionError.class,
+        () -> HostColumnVector.fromBigDecimals(new BigDecimal("12345678901234567890")));
+  }
+
+  @Test
+  public void testDecimalSpecifics() {
+    try (HostColumnVector cv = HostColumnVector.fromBigDecimals(decimal32Zoo)) {
+      assertEquals(DType.DTypeEnum.DECIMAL32, cv.getType().typeId);
+      assertEquals(-4, cv.getType().getScale());
+      assertFalse(cv.hasNulls());
+      assertEquals(decimal32Zoo[0], cv.getBigDecimal(0));
+      assertEquals(decimal32Zoo[1], cv.getBigDecimal(1));
+      assertEquals(decimal32Zoo[2], cv.getBigDecimal(2));
+    }
+    try (HostColumnVector cv = HostColumnVector.fromBigDecimals(decimal64Zoo)) {
+      assertEquals(DType.DTypeEnum.DECIMAL64, cv.getType().typeId);
+      assertEquals(-10, cv.getType().getScale());
+      assertTrue(cv.hasNulls());
+      assertEquals(decimal64Zoo[0], cv.getBigDecimal(0));
```

Comment:
Similar "make this test driven by input contents rather hardcoded" comment here.